internal/handlers/pg/pgdb/database_metadata.go

```diff
@@ -193,7 +256,29 @@ func (m *metadata) get(ctx context.Context, forUpdate bool) (*types.Document, er
 //
 // To avoid data race, set should be called only after getMetadata with forUpdate = true is called,
 // so that the metadata table is locked correctly.
-func (m *metadata) set(ctx context.Context, doc *types.Document) error {
+func (m *metadataStorage) set(ctx context.Context, metadata *metadata) error {
+	indexesArr := types.MakeArray(len(metadata.indexes))
+
+	for _, idx := range metadata.indexes {
+		keyDoc := types.MakeDocument(len(idx.key))
+		for _, pair := range idx.key {
+			keyDoc.Set(pair.Field, int32(pair.Order)) // order is set as int32 to be pjson-marshaled correctly
+		}
+
+		indexesArr.Append(must.NotFail(types.NewDocument(
+			"pgindex", idx.pgindex,
+			"name", idx.name,
+			"key", keyDoc,
+			"unique", idx.unique,
+		)))
+	}
+
+	doc := must.NotFail(types.NewDocument(
+		"_id", m.collection,
+		"table", metadata.pgtable,
```

Comment:
Please extract metadata-to-document and document-to-metadata functions, as there are multiple places where we do that

Reply:
fixed

Reply:
Where? We use `types.NewDocument` with "table" field there and in `metadata.ensure`. If there is metadata-to-document function, it should be called in both places instead

Reply:
Well, if you want me to create a structure and immediately convert it to a document, I can do it too, of course.

Reply:
fixed